### DISCUSSION TITLE
Towards version 1.0.0

### DISCUSSION THREAD

**Discussion by pete4abw** [pete4abw/lrzip-next#97] [Ideas]
I don't know about you, but I am tired of all these pre-release version numbers. `lrzip-next` has been in development for three years now and fifteen years since I began contributing to `lrzip`. Hardly any bugs are found. Performance continues to improve - especially with the **bzip3** backend addition. The git log is a jumbled mess with so many little changes.

So the plan is:
1. Merge **bzip3** onto the **lzma-22.01** branch
2. Give it a month and shake it out
3. Export the entire log and archive it
4. After that time, I will create an orphaned branch (`git checkout --orphan`) with all the latest changes and Release Tag it **v1.0.0**.
This means there will be no past log entries and the new master branch will look like a fresh clean new project.

**Comment by pete4abw** [comment]
@EDK-Rise, why the thumbs down? What part of the plan don't you like?

**Comment by pete4abw** [comment]
`lrzip-next` output during compression and decompression is a jumbled mess. A lot of the output was really meant for debugging purposes and to track progress -- Especially in the early days. I think one way to improve it is to insert `#ifdef DEBUG` statements to only show extreme output when debugging. Otherwise some cleanup is really needed.

This is something that needs to be dealt with prior to releasing any version 1.0.

**Comment by flawrist** [comment]
Dear developers, I'm a mere Windows 7 x64 user who is passionate about optimizations, including compression area. Is there a way for me to try `lrzip-next`? So far I see no pre-compiled binary for this OS.

**Comment by pete4abw** [comment]
> Dear developers, I'm a mere Windows 7 x64 user who is passionate about optimizations, including compression area. Is there a way for me to try `lrzip-next`? So far I see no pre-compiled binary for this OS.

Thank you for your comment. That is correct @sergeevabc . I removed all options for building under Windows/cygwin/mingw. `lrzip-next` is designed only to run under linux, with the best optimizations in x86_64. May I suggest you review `7-zip` which uses the lzma backend. See https://www.7-zip.org/ and https://sourceforge.net/projects/sevenzip/ for more info.
